### PR TITLE
annotationManager - grafana-limit flag added, token read from file feature added

### PR DESCRIPTION
**Pull request by indrarahul2013**

### CODE REVIEW

File: src/go/MONIT/annotationManager.go [hunk 1]
```diff
@@ -850,6 +865,7 @@ func parseTimes(ifCreatingAnnotation bool) []int64 {
 func main() {
 
 	flag.StringVar(&annotation, "annotation", "", "Annotation text")
+	flag.StringVar(&grafanaLimit, "grafana-limit", "100", "Limit for fetching number of annotations at a time.")
```
Comment (vkuznet) [inline src/go/MONIT/annotationManager.go:868]: the limit is an integer type, once you need to pass it to a string you'll use %d in fmt.Sprinf

File: src/go/MONIT/annotationManager.go [hunk 2]
```diff
@@ -40,6 +40,9 @@ var annotation string
 // annotation ID for deleting single annotation
 var annotationID int
 
+// limit for fetching number of annotations at a time
+var grafanaLimit string
```
Comment (vkuznet) [inline src/go/MONIT/annotationManager.go:44]: the limit is an integer data-type, please do not mix data types when it is not necessary

File: src/go/MONIT/annotationManager.go [hunk 3]
```diff
@@ -117,6 +120,8 @@ func getAnnotations(data *[]annotationData, tags []string) {
 		v.Add("tags", strings.Trim(tag, " "))
 	}
 
+	v.Add("limit", strings.Trim(grafanaLimit, " "))
```
Comment (vkuznet) [inline src/go/MONIT/annotationManager.go:123]: there is no need for Trim once you convert grafanaLimit to int, just use fmt.Sprintf("%d", grafanaLimit)

**Review by vkuznet — CHANGES_REQUESTED** [review]

File: src/go/MONIT/annotationManager.go (hunk 1 above)
Comment (indrarahul2013) [reply src/go/MONIT/annotationManager.go:868]: fixed.

File: src/go/MONIT/annotationManager.go (hunk 2 above)
Comment (indrarahul2013) [reply src/go/MONIT/annotationManager.go:44]: fixed.

File: src/go/MONIT/annotationManager.go (hunk 3 above)
Comment (indrarahul2013) [reply src/go/MONIT/annotationManager.go:123]: fixed.

**Pull request merged**